src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -0,0 +1,279 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "ObjectCacheStore.h"
+#include "Utils.h"
+#include <experimental/filesystem>
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_immutable_obj_cache
+#undef dout_prefix
+#define dout_prefix *_dout << "ceph::cache::ObjectCacheStore: " << this << " " \
+                           << __func__ << ": "
+
+namespace efs = std::experimental::filesystem;
+
+namespace ceph {
+namespace immutable_obj_cache {
+
+ObjectCacheStore::ObjectCacheStore(CephContext *cct)
+      : m_cct(cct), m_rados(new librados::Rados()),
+        m_ioctx_map_lock("ceph::cache::ObjectCacheStore::m_ioctx_map_lock") {
+
+  object_cache_max_size =
+    m_cct->_conf.get_val<Option::size_t>("immutable_object_cache_max_size");
+
+  std::string cache_path =
+    m_cct->_conf.get_val<std::string>("immutable_object_cache_path");
+  m_cache_root_dir = cache_path + "/ceph_immutable_obj_cache/";
+
+  //TODO(): allow to set cache level
+  m_policy = new SimplePolicy(m_cct, object_cache_max_size/(4096*1024), 0.1);
```

Comment:
You cannot assume 4MiB objects -- that's really only the default for RBD images but it's not the default for RGW. If we cannot add space usage tracking (should be straightforward to add to the policy), perhaps the configuration option needs to be just max objects.

Reply:
yes, should update this counter on promotion_callback, will do this in the next commit

Reply:
@dillaman  fix pushed. SimplePolicy will maintain its own counter on each promotion/eviction done.